src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/CommandProcessor.java

```diff
@@ -1773,20 +1773,49 @@ public void doit(Tokens t) {
                 sysProps.run();
             }
         },
-        new Command("dumpheap", "dumpheap [filename]", false) {
+        new Command("dumpheap", "dumpheap [gz=<1-9>] [filename]", false) {
             public void doit(Tokens t) {
-                if (t.countTokens() > 1) {
+                if (t.countTokens() > 2) {
                     usage();
                 } else {
                     JMap jmap = new JMap();
-                    String filename;
-                    if (t.countTokens() == 1) {
-                        filename = t.nextToken();
+                    String filename = "heap.bin";
+                    int gzlevel = 0;
+                    /* Parse "gz=" option. */
+                    String option = t.nextToken();
+                    String[] keyValue = option.split("=");
+                    if (keyValue.length == 1) {
+                        err.println("Argument is expected for \"gz\"");
+                        usage();
+                        return;
+                    }
+                    if (keyValue[0].equals("gz")) {
```

Comment:
This code now assumes there will be a `gz` option specified. If there isn't, you will still execute this section of code, and get an NPE at this point.